config/notifiers.go

```diff
@@ -459,17 +460,38 @@ type OpsGenieConfig struct {
 	Description string            `yaml:"description,omitempty" json:"description,omitempty"`
 	Source      string            `yaml:"source,omitempty" json:"source,omitempty"`
 	Details     map[string]string `yaml:"details,omitempty" json:"details,omitempty"`
-	Teams       string            `yaml:"teams,omitempty" json:"teams,omitempty"`
-	Tags        string            `yaml:"tags,omitempty" json:"tags,omitempty"`
-	Note        string            `yaml:"note,omitempty" json:"note,omitempty"`
-	Priority    string            `yaml:"priority,omitempty" json:"priority,omitempty"`
+	// Teams field is deprecated.
+	Teams      string                    `yaml:"teams,omitempty" json:"teams,omitempty"`
+	Responders []OpsGenieConfigResponder `yaml:"responders,omitempty" json:"responders,omitempty"`
+	Tags       string                    `yaml:"tags,omitempty" json:"tags,omitempty"`
+	Note       string                    `yaml:"note,omitempty" json:"note,omitempty"`
+	Priority   string                    `yaml:"priority,omitempty" json:"priority,omitempty"`
 }
 
 // UnmarshalYAML implements the yaml.Unmarshaler interface.
 func (c *OpsGenieConfig) UnmarshalYAML(unmarshal func(interface{}) error) error {
 	*c = DefaultOpsGenieConfig
 	type plain OpsGenieConfig
-	return unmarshal((*plain)(c))
+	if err := unmarshal((*plain)(c)); err != nil {
+		return err
+	}
+
+	if c.Teams != "" {
+		// Should we allow this for backward compatibility? Should we fill responders from this?
```

Comment:
Given that we can't propagate the deprecation to the user in the logs, I think that a hard failure is preferred.